src/Shared/Components/CICDHistory/DeploymentHistoryConfigDiff/DeploymentHistoryConfigDiff.tsx

```diff
@@ -54,32 +56,49 @@ export const DeploymentHistoryConfigDiff = ({
 
     // ASYNC CALLS
     // Load comparison deployment data
-    const [compareDeploymentConfigLoader, compareDeploymentConfig, , reloadCompareDeploymentConfig] = useAsync(
-        () =>
-            Promise.allSettled([
-                getAppEnvDeploymentConfig({
-                    params: {
-                        appName,
-                        envName,
-                        configArea: 'DeploymentHistory',
-                        pipelineId,
-                        wfrId: currentWfrId,
-                    },
-                }),
+    const [compareDeploymentConfigLoader, compareDeploymentConfig, , reloadCompareDeploymentConfig] =
+        useAsync(async () => {
+            const payloads = [
+                {
+                    appName,
+                    envName,
+                    configArea: 'DeploymentHistory',
+                    pipelineId,
+                    wfrId: currentWfrId,
+                } as const,
                 isPreviousDeploymentConfigAvailable
-                    ? getAppEnvDeploymentConfig({
-                          params: {
-                              appName,
-                              envName,
-                              configArea: 'DeploymentHistory',
-                              pipelineId,
-                              wfrId: compareWfrId,
-                          },
-                      })
+                    ? ({
+                          appName,
+                          envName,
+                          configArea: 'DeploymentHistory',
+                          pipelineId,
+                          wfrId: compareWfrId,
+                      } as const)
                     : null,
-            ]),
-        [currentWfrId, compareWfrId],
-    )
+            ] as const
+
+            const [secretsData, ...configDatas] = await Promise.allSettled([
+                !isSuperAdmin ? getCompareSecretsData([...payloads]) : null,
```

Comment:
remove this check as well

Reply:
As per discussions with @vivek-devtron , if we remove this check, for superAdmins there will redundant api calls. The probability of a person being demoted from super admin to view only user is very slim. So we can expect that the user will at-least refresh the page.